conftest.py

```diff
@@ -127,6 +131,55 @@ def pytest_keyboard_interrupt(excinfo):
         exp.stop()
 
 
+@pytest.fixture
+def app_id(request):
+    return ttn.APP_ID
+
+
+@pytest.fixture
+def dev_id(request):
+    if request.param == "otaa":
+        return ttn.DEVICE_ID
+
+    return ttn.DEVICE_ID_ABP
+
+
+@pytest.fixture
+def appkey(request):
+    return get_required_envvar("LORAWAN_APP_KEY")
+
+
+@pytest.fixture
+def appeui(request):
+    return ttn.APPEUI
+
+
+@pytest.fixture
+def deveui(request):
+    return ttn.DEVEUI
+
+
+@pytest.fixture
+def devaddr(request):
+    return ttn.DEVADDR
+
+
+@pytest.fixture
+def nwkskey(request):
+    return get_required_envvar("LORAWAN_NWK_SKEY")
+
+
+@pytest.fixture
+def appskey(request):
+    return get_required_envvar("LORAWAN_APP_SKEY")
+
+
+@pytest.fixture
+def mqtt_descriptor(request):
```

Comment:
Does the name `ttn_client` make maybe make more sense now? Also regarding naming `client.client` should probably be called `client.mqtt_client` now.

Reply:
done

Reply:
ah, I forgot to change the name from client to mqtt_client...

Reply:
maybe rather `ttn_client` then ;-) This isn't generating a general MQTT client.

Reply:
done